enhancements/installer/aws-lb-subnet-selection.md

```diff
@@ -0,0 +1,686 @@
+
+1. Deprecate the `platform.aws.subnets` field in the install-config and replace with
+   `platform.aws.subnetsConfig`, which more intuitively manages both cluster and ingress subnets selection.
+2. Add the `Subnets` field to `spec.loadBalancer.platform.aws.subnets` in the IngressConfig
+   to encode the default subnets for IngressControllers.
+
+The proposal to deprecate the `platform.aws.subnets` field in the install-config is driven by
+its ambiguity, as it does not clearly distinguish between ingress subnets and cluster
+subnets, which can be different. Additionally, `platform.aws.subnets` is a `[]string` and doesn't
+allow for future expansion or adaptation for other subnet type, roles, or metadata information.
+
+The IngressConfig holds the default values that the Ingress Operator will use when creating
+the default IngressController, as well as any new IngressController. The IngressConfig will be
+populated by the installer, similar to how the [Allow Users to specify Load Balancer Type during installation for AWS](/enhancements/installer/aws-load-balancer-type.md)
+enhancement handled `spec.loadBalancer.platform.aws.lbType`.
+
+The OpenShift Installer will be updated to support the new `platform.aws.subnetsConfig` field, and the
+Ingress Operator will be updated to apply the `Subnets` field in the IngressConfig as the default subnets
+for new IngressControllers.
+
+### Implementation Details/Notes/Constraints
+
+As mentioned in [Proposal](#proposal), the install-config and the IngressConfig will be updated
+to support propagating default subnets to all new IngressControllers, including the default
+IngressController. The Ingress Operator will also need to be updated to use the new IngressConfig
+`Subnet` field.
+
+#### Installer Updates
+
+The `platform.aws.subnets` field in the install-config will be deprecated and replaced with a new
+field, `platform.aws.subnetsConfig`. Much like `Subnets`, the new `SubnetsConfig` field indicates
+a list of subnets in a pre-existing VPC, but also provide the role that the subnet will fulfill in the
+cluster. Additionally, since it is a struct and not a `[]string`, it provide the ability to be
+expanded with additional subnet-related fields in the future.
+
+```go
+// Platform stores all the global configuration that all machinesets
+// use.
+// +kubebuilder:validation:XValidation:rule=`!(has(self.subnets) && has(self.subnetsConfig))`,message="cannot use both subnets and subnetsConfig"
+type Platform struct {
+	[...]
+    // Subnets specifies existing subnets (by ID) where cluster
+    // resources will be created.  Leave unset to have the installer
+    // create subnets in a new VPC on your behalf.
+	//
+    // Deprecated: Use SubnetsConfig
+    //
+    // +optional
+    Subnets []string `json:"subnets,omitempty"`
+
+	// SubnetsConfig specifies the subnet configuration for
+	// a cluster by specifying a list of subnet ids with their
+	// designated role. At least one Cluster role subnet must be
```

Comment:
I'll come back to this after hearing more about the inclusion of Cluster subnets in the proposal.